fetching/repo.py

```diff
@@ -0,0 +1,112 @@
+from abc import ABC, abstractmethod
+
+import logging
+import requests
+import pathlib
+import tempfile
+import typing
+import shutil
+
+import git
+import tqdm # type: ignore
+
+from .projio import Project
+
+
+class Repository(ABC):
+    def __init__(self, project_url: str):
+        self.project_url = project_url
+
+    def fetch(self, output: pathlib.Path) -> Project:
+        """
+        Download the project from the URL specified in the constructor
+        and store it in the path specified by output
+        """
+        td = self._fetch()
+        self._write_to(td, output)
+
+        return Project(output)
+
+    @property
+    @abstractmethod
+    def fmt(self) -> str:
+        """
+        Static attribute indicating format of repository
```

Comment:
Static is the wrong word. Use something else.

Reply:
Updated.